src/npc.cpp

```diff
@@ -1811,13 +1809,11 @@ int npc::value( const item &it, int market_price ) const
 
     // TODO: Artifact hunting from relevant factions
     // ALSO TODO: Bionics hunting from relevant factions
-    if( fac_has_job(FACJOB_DRUGS) && it.is_food() &&
-        (dynamic_cast<const it_comest*>(it.type))->addict >= 5 ) {
+    if( fac_has_job(FACJOB_DRUGS) && it.is_food() && it.type->comestible->addict >= 5 ) {
         ret += 10;
     }
 
-    if( fac_has_job(FACJOB_DOCTORS) && it.is_food() &&
-        dynamic_cast<const it_comest*>( it.type )->comesttype == "MED" ) {
+    if( fac_has_job(FACJOB_DOCTORS) && it.is_food() && it.type->comestible->comesttype == "MED" ) {
```

Comment:
While it is good for compatibility and ease of porting, in the future meds probably shouldn't be checked for with `is_food`, but `is_med`.